backend/accounts.go

```diff
@@ -1722,6 +1722,12 @@ func (backend *Backend) checkAccountUsed(account accounts.Interface) {
 			return
 		}
 	}
+
+	// If the account is already marked as used, nothing to do.
+	if account.Config().Config.Used {
+		return
```

Comment:
feels safer to continue with the code below, i.e. set HiddenBecauseUnused = true, and make the `maybeAddHiddenUnusedAccounts()` and emit event calls.

haven't checked all implications exactly, but by returning here the behavior might change because it is not doing the same thing anymore.